GVFS/GVFS.Common/GitHubUpgrader.cs

```diff
@@ -238,11 +246,9 @@ protected virtual bool TryDownloadAsset(Asset asset, out string errorMessage)
             errorMessage = null;
 
             string downloadPath = ProductUpgraderInfo.GetAssetDownloadsPath();
-            Exception exception;
-            if (!this.fileSystem.TryCreateDirectory(downloadPath, out exception))
+            if (!this.fileSystem.DirectoryExists(downloadPath))

```

Comment:
In this case - whose responsibility it is to create the download directory?

I had approached this as the Download method would ensure the download directory exists and create it if necessary.

Reply:
I went back and forth on how\where to make this change.

With these changes it's the responsibility of the `ProductUpgradeTimer` (in GVFS.Service) and the `UpgradeOrchastrator` (in the `gvfs` verb).

My reasoning was:

- Creating the download directory is a common task (for both upgraders), and I wanted to create the directory in a common spot for both.

- Avoid growing the scope of `TryDownloadAsset` (now that we need to set ACLs as well)

Another alternative I considered was making the call in both upgrader's versions of `TryDownloadNewestVersion`.

@jamill I could not figure out where this directory was getting created for the NuGet upgrader, do you know if the nuget library was creating it if it didn't exist?

Reply:
> @jamill I could not figure out where this directory was getting created for the NuGet upgrader, do you know if the nuget library was creating it if it didn't exist?

Yes, the NuGet library was creating it as part of downloading the package if it didn't exist.

I like having the `Upgraders` `TryDownloadNewestVersion` being responsible for creating the directory if it doesn't exist, as this will remove an interaction / expectation between the UpgradeOrchestrator and the Upgraders. The fact that the NuGet library was creating the directory was covering for the upgrader missing this logic in the 1st place.